### PR TITLE
creating dataclass wrapper around config variables

### PR DESCRIPTION
**Pull request by andrew-rti**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,5 +28,7 @@ WORKDIR $HOME/dug
 
 RUN make install
 
+USER $USER
```
Comment (andrew-rti) [inline Dockerfile:31]: Was running into permissions issues wrt `chown` during `docker-compose build` and moving the location resolved the issue.

File: src/dug/core.py [hunk 2]
```diff
@@ -42,7 +41,7 @@ def __init__(self, message, details):
 
 
 class ParserNotFoundException(Exception):
-    pass
+    ...
```
Comment (andrew-rti) [inline src/dug/core.py:44]: changed this to ellipsis to be consistent with `def status(self):` at the bottom of the file

File: tests/integration/test_parsers.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from dug.parsers import DbGaPParser, TOPMedTagParser
+from src.dug.parsers import DbGaPParser, TOPMedTagParser
```
Comment (andrew-rti) [inline tests/integration/test_parsers.py:1]: this ended up being the fix for me when running `pytest .`... not sure how i feel about it.. I mean, it makes sense from a pathing standpoint.. but still. How was this working before?

File: src/dug/config.py [hunk 4]
```diff
@@ -54,9 +47,50 @@
...
+        "elastic_host": "ELASTIC_API_HOST",
+        "elastic_port": "ELASTIC_API_PORT",
+        "elastic_username": "ELASTIC_USERNAME",
+        "elastic_password": "ELASTIC_PASSWORD",
+        "redis_host": "REDIS_HOST",
+        "redis_port": "REDIS_PORT",
+        "redis_password": "REDIS_PASSWORD",
+        "nboost_host": "NBOOST_API_HOST",
+        "nboost_port": "NBOOST_API_PORT"
+    }
```
Comment (andrew-rti) [inline src/dug/config.py:88]: This was the part where I wasn't entirely satisfied with how it's implemented. Basically this mapping exists so i can easily go over the relevant environment variables in a loop and build up the `kwargs` to be expanded later. Open to any/all suggestions.

Comment (cschreep) [reply src/dug/config.py:88]: I think it's fine to leave it here. These should be the only places these environment variables are referenced now. The only other improvement might be to make them defined as constants somewhere. I do think the from_env function should be a classmethod on Config though

File: src/dug/core.py [hunk 5]
```diff
@@ -55,34 +54,26 @@ class Search:
          * disease->study
          * disease->phenotype->study
     """
+    def __init__(self, cfg=config.from_env(), indices=['concepts_index', 'variables_index', 'kg_index']):
```
Comment (cschreep) [inline src/dug/core.py:57]: I don't think we should have a default value for the config here, I think we should make it a required param (and update other usages accordingly)


```suggestion
    def __init__(self, cfg: Config, indices=['concepts_index', 'variables_index', 'kg_index']):
```

File: src/dug/core.py [hunk 6]
```diff
@@ -616,12 +605,8 @@ class Dug:
...
     kg_index = "kg_index"
 
-    def __init__(self):
-        self._search = Search(
-            host=cfg.elasticsearch_host,
-            port=cfg.elasticsearch_port,
-            indices=[self.concepts_index, self.variables_index, self.kg_index]
-        )
+    def __init__(self, cfg=config.from_env()):
+        self._search = Search(indices=[self.concepts_index, self.variables_index, self.kg_index])
```
Comment (cschreep) [inline src/dug/core.py:609]: we don't do anything with the cfg param here, and I think it should be a required param as well (and the usages updated accordingly)

```suggestion
    def __init__(self, cfg: Config):
        self._cfg = cfg
        self._search = Search(cfg, indices=[self.concepts_index, self.variables_index, self.kg_index])
```

File: tests/integration/test_parsers.py (hunk 3 above)
Comment (cschreep) [reply tests/integration/test_parsers.py:1]: You shouldn't have to include src in the import path, that suggests your environment doesn't have dug in the path

File: src/dug/config.py [hunk 7]
```diff
@@ -54,9 +47,50 @@
...
+
+    kwargs = {}
+
+    for kwarg, env_var in env_vars.items():
+        if env_value := os.environ.get(env_var):
```
Comment (cschreep) [inline src/dug/config.py:93]: Jenkins tests failed since we support Python 3.7, and the `:=` operator isn't available in that version

**Review by cschreep — CHANGES_REQUESTED** [review]
Looks good, just a couple adjustments requested

**Review by cschreep — APPROVED** [review]

**Pull request merged**